Mixpanel/Mixpanel.m

```diff
@@ -25,21 +25,21 @@
 
 @implementation Mixpanel
 
-static Mixpanel *sharedInstance;
+static NSMutableDictionary *instances;
+
 + (Mixpanel *)sharedInstanceWithToken:(NSString *)apiToken launchOptions:(NSDictionary *)launchOptions
 {
-    static dispatch_once_t onceToken;
-    dispatch_once(&onceToken, ^{
-        
+    if (instances && [instances objectForKey:apiToken]) {
```

Comment:
i think `dictionary[key]` is standard instead of `objectForKey`. also you don't need to check if `instances` because if it doesn't exist `objectForKey` will be nil by default